### PR TITLE
Enable/disable CalDAV calendar synchronization

### PR DESCRIPTION
**Pull request by bertrandda**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Add the ability to choose which calendars to synchronize in CalDAV integration.

image
image

### CODE REVIEW

File: server/test/lib/calendar/calendar.event.test.js [hunk 1]
```diff
@@ -49,6 +49,13 @@ describe('calendar.destroy', () => {
   });
 });
 
+describe('calendar.destroyEvents', () => {
+  const calendar = new Calendar();
+  it("should destroy all calendar's event", async () => {
+    await calendar.destroyEvents('0dc03aef-4a23-9c4e-88e3-5437971269e5');
```
Comment (atrovato) [inline server/test/lib/calendar/calendar.event.test.js]: No assertion?

**Review by atrovato — DISMISSED** [review]
Looks good to my eyes, but didn't try with real calendars. Please find any testers to confirm it works well.

**Review by atrovato — DISMISSED** [review]
Good for me, but didn't test in real life.

**Comment by VonOx** [comment]
Hi @bertrandda 

I've tested this PR with Google calendar

No error in logs, but no event in Gladys Calendar ( Waited from 2H to be sure )
```
2021-03-13T11:07:15+0100 <info> calendar.cleanUp.js:12 (CalDAVHandler.cleanUp) Start cleaning CalDAV data for user: 8da2b78a-bbbb-4804-b39c-a0a39ed0ca66
2021-03-13T11:07:15+0100 <info> calendar.cleanUp.js:14 (CalDAVHandler.cleanUp) 0 calendars to clean
2021-03-13T11:07:30+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar XXXXXXX@gmail.com
2021-03-13T11:07:30+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar 4CAD
2021-03-13T11:07:30+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar Anniversaires
2021-03-13T11:07:30+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar Jours fériés en France
2021-03-13T11:07:30+0100 <info> calendar.syncUserCalendars.js:38 (CalDAVHandler.syncUserCalendars) CalDAV : Found 4 calendars.
2021-03-13T11:07:34+0100 <info> calendar.syncUserCalendars.js:132 (Promise.map.concurrency) CalDAV : 167 events updated for calendar XXXXXXX@gmail.com.
2021-03-13T11:07:35+0100 <info> calendar.syncUserCalendars.js:132 (Promise.map.concurrency) CalDAV : 57 events updated for calendar Jours fériés en France.
2021-03-13T11:07:37+0100 <info> calendar.syncUserCalendars.js:132 (Promise.map.concurrency) CalDAV : 45 events updated for calendar Anniversaires.
2021-03-13T11:07:42+0100 <info> calendar.syncUserCalendars.js:132 (Promise.map.concurrency) CalDAV : 250 events updated for calendar 4CAD.
2021-03-13T11:07:48+0100 <debug> bluetooth.scanPresence.js:24 (BluetoothManager.scanPresence) Bluetooth: 0 presence sensor features
2021-03-13T11:08:03+0100 <info> calendar.enableCalendar.js:19 (CalDAVHandler.enableCalendar) Calendar jours-feries-en-france disabled & emptied
2021-03-13T11:08:03+0100 <info> calendar.enableCalendar.js:19 (CalDAVHandler.enableCalendar) Calendar anniversaires disabled & emptied
2021-03-13T11:08:26+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar XXXXXXX@gmail.com
2021-03-13T11:08:26+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar 4CAD
2021-03-13T11:08:26+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar Anniversaires
2021-03-13T11:08:26+0100 <info> calendar.requests.js:52 () CalDAV : Found calendar Jours fériés en France
2021-03-13T11:08:26+0100 <info> calendar.syncUserCalendars.js:38 (CalDAVHandler.syncUserCalendars) CalDAV : Found 4 calendars.
```

**Comment by bertrandda** [comment]
@VonOx There is no error in your logs & I just tried with Google calendar, it seems to work on my side. Can you check your `t_calendar_event` table in the database ?
To avoid waiting you can force synchronisation in Integrations -> Account -> Sync

**Comment by VonOx** [comment]
Seem's to be a problem with my instance/google calendar.

PR is ok !

**Review by VonOx — DISMISSED** [review]

File: server/services/caldav/api/caldav.controller.js [hunk 2]
```diff
@@ -38,6 +38,16 @@ module.exports = function CalDAVController(caldavHandler) {
     });
   }
 
+  /**
+   * @api {patch} /api/v1/service/caldav/enable Enable or disable calendar synchronization
+   * @apiName Enable
+   * @apiGroup CalDAV
+   */
+  async function enable(req, res) {
```
Comment (Pierre-Gilles) [inline server/services/caldav/api/caldav.controller.js:46]: It's weird to have an "enable" function that either enable or disable

Either you have a PATCH calendar API call (simple), or either you have a "enable" + "disable" route.

Pick one :)

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
See my comments :)

File: server/services/caldav/api/caldav.controller.js (hunk 2 above)
Comment (bertrandda) [reply server/services/caldav/api/caldav.controller.js:46]: I chose to split in 2 routes because we need actions after disabling

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Looks good to me! 

Just one change, I just merged this PR: https://github.com/GladysAssistant/Gladys/pull/1133

And we can no longer use this.context.intl.dictionary in the front, but instead we use a HoC component to inject the dictionnary as props :) 

Could you make the change in this PR? (see the PR to see how to do it, it's pretty easy)

Thanks for your PR!

**Review by atrovato — COMMENTED** [review]
Code is great enough for me.

**Review by Pierre-Gilles — APPROVED** [review]
Tested locally, works fine!

**Pull request merged**